include/fuzzuf/cli/fuzzer/mopt/build_mopt_fuzzer_from_args.hpp

```diff
@@ -0,0 +1,243 @@
+/*
+ * fuzzuf
+ * Copyright (C) 2021 Ricerca Security
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+#ifndef FUZZUF_INCLUDE_CLI_MOPT_BUILD_MOPT_FROM_ARGS_HPP
+#define FUZZUF_INCLUDE_CLI_MOPT_BUILD_MOPT_FROM_ARGS_HPP
+
+#include <boost/program_options.hpp>
+#include <memory>
+
+#include "fuzzuf/algorithms/afl/afl_option.hpp"
+#include "fuzzuf/algorithms/mopt/mopt_optimizer.hpp"
+#include "fuzzuf/algorithms/mopt/mopt_option.hpp"
+#include "fuzzuf/algorithms/mopt/mopt_state.hpp"
+#include "fuzzuf/cli/fuzzer_args.hpp"
+#include "fuzzuf/cli/global_fuzzer_options.hpp"
+#include "fuzzuf/cli/put_args.hpp"
+#include "fuzzuf/exceptions.hpp"
+#include "fuzzuf/executor/linux_fork_server_executor.hpp"
+#include "fuzzuf/executor/native_linux_executor.hpp"
+#include "fuzzuf/executor/qemu_executor.hpp"
+#include "fuzzuf/optimizer/optimizer.hpp"
+#include "fuzzuf/utils/common.hpp"
+#include "fuzzuf/utils/optparser.hpp"
+#include "fuzzuf/utils/workspace.hpp"
+
+namespace fuzzuf::cli::fuzzer::mopt {
+
+namespace po = boost::program_options;
+
+struct MOptFuzzerOptions {
+  bool forksrv;                        // Optional
+  std::vector<std::string> dict_file;  // Optional
+  bool frida_mode;                     // Optional
+  u64 mopt_limit_time = 1;
+  u64 mopt_most_time = 0;
+
+  // Default values
+  MOptFuzzerOptions() : forksrv(true), frida_mode(false){};
```

Comment:
I would like this part to be unified with default initialization or without it.

Reply:
fixed in 0a7be14649ee0df3dda4605efe02a2b03918f748